File: LeetCode/MaxDepth/MaxDepth.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def maxDepth(self, root) -> int:
        if root is None:
            return 0

        lDepth = self.maxDepth(root.left)
        rDepth = self.maxDepth(root.right)

        if (lDepth > rDepth):
            return lDepth+1
        return rDepth+1

root = TreeNode(3)
root.left = TreeNode(9)
root.right = TreeNode(20)
root.right.left = TreeNode(15)
root.left.right = TreeNode(7)

solu = Solution()

print(solu.maxDepth(root))

File: LeetCode/MaxDepth/MaxDepth.java
class Node{
   
    int val;
    Node left;
    Node right;
    Node(){}
    
    Node(int val){
        this.val = val;
    }
    
    Node(int val, Node left, Node right){
        this.val = val;
        this.left = left;
        this.right = right;
    }
}



class Solution{
    public int maxDepth(Node root){
        if (root == null){
            return 0;
        }

        int lDepth = maxDepth(root.left);

        int rDepth = maxDepth(root.right);

        if (lDepth > rDepth){
            return (++lDepth);
        }
        
        return (++rDepth);
    }

}